### PR TITLE
Use AzDO Pipeline Resources to define our build Docker containers

### PR DESCRIPTION
**Pull request by jkoritzinsky**
Refactor all of the docker images we use in our build into AzDO container resources.

Also update all of our pipelines (we have a lot) to use this new format.

This enables a few things:

1. We can pass the ROOTFS_DIR environment variable as part of the container.
2. Docker container names are better centralized in an easier to read and easier to parse file.
3. Reduce duplication of defining container resources for containers used in individual steps (instead of job-wide).
4. Enable an easy mechanism to reference specific containers that we use job-wide in specific steps.

The vast majority of changes in this PR are whitespace changes. I highly recommend viewing changes either side-by-side (to validate YAML whitespace changes) or with whitespace changes hidden (for the more interesting logic changes in the YAML).

### CODE REVIEW

File: eng/pipelines/runtime.yml [hunk 1]
```diff
@@ -51,1248 +51,1234 @@ pr:
...
+        - windows_x86
+        - windows_x64
+        - windows_arm
+        - windows_arm64
+        jobParameters:
+          testGroup: innerloop
+          condition: >-
+            or(
+              eq(dependencies.evaluate_paths.outputs['SetPathVars_coreclr.containsChange'], true),
+              eq(variables['nonWasmRuntimeTestsContainsChange'], true),
```
Comment (akoeplinger) [inline eng/pipelines/runtime.yml:93]: could you please double check the changes in this file? It looks like these are wrong merges.

File: eng/pipelines/coreclr/perf_slow.yml [hunk 2]
```diff
@@ -27,231 +27,251 @@ schedules:
...
+          - Linux_arm64
+          jobParameters:
+            testGroup: perf
+            liveLibrariesBuildConfig: Release
+            runtimeType: mono
+            codeGenType: 'Interpreter'
+            projectFile: microbenchmarks.proj
+            runKind: micro_mono
+            runJobTemplate: /eng/pipelines/coreclr/templates/run-performance-job.yml
+            logicalmachine: 'perfa64'
```
Comment (akoeplinger) [inline eng/pipelines/coreclr/perf_slow.yml:74]: is this intentional?

File: eng/pipelines/common/templates/pipeline-with-resources.yml [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+  - name: stages
+    type: stageList
+
+resources:
+  containers:
+    - container: Linux_arm
+      image: mcr.microsoft.com/dotnet-buildtools/prereqs:ubuntu-18.04-cross-arm-20220907130538-70ed2e8
+      env:
+        ROOTFS_DIR: /crossrootfs/arm
+    - container: Linux_armv6
```
Comment (akoeplinger) [inline eng/pipelines/common/templates/pipeline-with-resources.yml:11]: nit: add newlines between the container definitions, it is quite hard to read as is

**Review by akoeplinger — APPROVED** [review]

**Review by hoyosjs — APPROVED** [review]
Do you have an official build?

**Comment by jkoritzinsky** [comment]
Official build at https://dev.azure.com/dnceng/internal/_build/results?buildId=2013530&view=results

**Comment by jkoritzinsky** [comment]
Files #76750 for the timeout

**Pull request merged**